sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -24,6 +24,7 @@ import scala.util.Try
 import org.apache.spark.annotation.Unstable
 import org.apache.spark.sql.catalyst.trees.SQLQueryContext
 import org.apache.spark.sql.errors.QueryExecutionErrors
+import org.apache.spark.sql.errors.SqlApiQueryExecutionErrors
```

Comment:
Do other similar items need to be moved? eg:
QueryExecutionErrors.unscaledValueTooLargeForPrecisionError(this, precision, scale)
It seems that it is only called in `DataType`.

Reply:
For those we haven't moved, they are using `SQLConf` inside and we are still discussing how to better deal with`SQLConf` now.